pkg/aws/elasticache.go

```diff
@@ -0,0 +1,137 @@
+	"github.com/aws/aws-sdk-go/service/elasticache"
+	"github.com/aws/aws-sdk-go/service/elasticache/elasticacheiface"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi"
+	"github.com/aws/aws-sdk-go/service/resourcegroupstaggingapi/resourcegroupstaggingapiiface"
+	"github.com/integr8ly/cluster-service/pkg/clusterservice"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"strings"
+)
+
+var _ ActionEngine = &ElasticacheEngine{}
+var TagFilters []*resourcegroupstaggingapi.TagFilter
+
+type ElasticacheEngine struct {
+	elasticacheClient elasticacheiface.ElastiCacheAPI
+	taggingClient     resourcegroupstaggingapiiface.ResourceGroupsTaggingAPIAPI
+	logger            *logrus.Entry
+}
+
+func NewDefaultElastiCacheEngine(session *session.Session, logger *logrus.Entry) *ElasticacheEngine {
+	return &ElasticacheEngine{
+		elasticacheClient: elasticache.New(session),
+		taggingClient:     resourcegroupstaggingapi.New(session),
+		logger:            logger.WithField("engine", "aws_elasticache"),
+	}
+}
+
+func (r *ElasticacheEngine) GetName() string {
+	return "AWS elasticache Engine"
+}
+
+//Delete all RDS resources for a specified cluster
+func (r *ElasticacheEngine) DeleteResourcesForCluster(clusterId string, tags map[string]string, dryRun bool) ([]*clusterservice.ReportItem, error) {
+	logger := r.logger.WithFields(logrus.Fields{"clusterId": clusterId, "dryRun": dryRun})
+	logger.Debug("deleting resources for cluster")
+
+	var reportItems []*clusterservice.ReportItem
+	var replicationGroupsToDelete []string
+	resourceInput := &resourcegroupstaggingapi.GetResourcesInput{
+		ResourceTypeFilters: aws.StringSlice([]string{"elasticache:cluster"}),
+		TagFilters: []*resourcegroupstaggingapi.TagFilter{
+			{
+				Key: aws.String(tagKeyClusterId),
+				Values: aws.StringSlice([]string{
+					clusterId,
+				}),
+			},
+		},
+	}
+	resourceOutput, err := r.taggingClient.GetResources(resourceInput)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to Get resources for clusterID : %s", clusterId)
+	}
+
+	for _, resourceTagMapping := range resourceOutput.ResourceTagMappingList {
+
+		arn := aws.StringValue(resourceTagMapping.ResourceARN)
+		arnSplit := strings.Split(arn, ":")
+		cacheClusterId := arnSplit[len(arnSplit)-1]
+		cacheClusterInput := &elasticache.DescribeCacheClustersInput{
+			CacheClusterId: aws.String(cacheClusterId),
+		}
+		cacheClusterOutput, err := r.elasticacheClient.DescribeCacheClusters(cacheClusterInput)
+		if err != nil {
+			return nil, errors.Wrapf(err, "cannot get cacheclusterOutput for : %s", cacheClusterInput)
+		}
+		for _, cacheCluster := range cacheClusterOutput.CacheClusters {
+			dbLogger := logger.WithField("replicationGroup", aws.String(*cacheCluster.ReplicationGroupId))
+			if contains(replicationGroupsToDelete, *cacheCluster.ReplicationGroupId) {
+				dbLogger.Debugf("Replication Group already exists in deletion list (%s=%s)", *cacheCluster.ReplicationGroupId, clusterId)
```

Comment:
Let's try to keep log and error messages lower case, apart from for variable names, to keep consistency